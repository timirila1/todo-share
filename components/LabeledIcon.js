import React from 'react';
import { StyleSheet, TouchableHighlight, Text, View } from 'react-native';
import { EvilIcons } from '@expo/vector-icons';
import { colors } from '../services/consts';

export const LabeledIcon = props => {
    const { onPress, label, name } = props;

    return (
        <TouchableHighlight
            onPress={onPress}
            style={styles.labeledIcon}
            underlayColor="rgba(255, 255, 255, 0)"
        >
            <View>
                <EvilIcons
                    style={styles.icon}
                    name={name}
                    size={36}
                    color={colors.additional}
                />
                <Text style={styles.text}>{label}</Text>
            </View>
        </TouchableHighlight>
    );
};

const styles = StyleSheet.create({
    labeledIcon: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    icon: {
        alignSelf: 'center',
    },
    text: {
        fontSize: 12,
        color: colors.textDark,
        textAlign: 'center'
    }
});
