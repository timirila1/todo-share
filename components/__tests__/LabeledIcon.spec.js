import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { LabeledIcon } from '../LabeledIcon';

it('Render LabeledIcon', async () => {
    const tree = renderer.create(
        <LabeledIcon
            label="Test"
            name="plus"
        />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
