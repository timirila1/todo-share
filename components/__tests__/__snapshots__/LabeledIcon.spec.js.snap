// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render LabeledIcon 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      false,
    ]
  }
>
  <View
    style={null}
  >
    <Text />
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.6)",
            "fontSize": 12,
            "textAlign": "center",
          },
          false,
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;
